### PR TITLE
More syscall fuzz test

### PR DESCRIPTION
**Pull request by ImTei**
**Description**

Following up on the previous PR(https://github.com/ethereum-optimism/asterisc/pull/21), added fuzz tests for rest of syscalls.
- FuzzStateSyscallBrk
- FuzzStateSyscallMmap
- FuzzStateSyscallFcntl
- FuzzStateSyscallOpenat
- FuzzStateSyscallClockGettime
- FuzzStateSyscallClone
- FuzzStateSyscallGetrlimit
- FuzzStateRead
  - stdin, errors
- FuzzStateWrite
  - stdout, stderr, errors

And fixed getrlimit syscall implementation to return 0 ([ref](https://man7.org/linux/man-pages/man2/getrlimit.2.html#RETURN_VALUE))

### CODE REVIEW

File: rvgo/test/syscall_test.go [hunk 1]
```diff
@@ -0,0 +1,903 @@
...
+		fastState := fast.NewInstrumentedState(state, nil, os.Stdout, os.Stderr)
+		stepWitness, err := fastState.Step(true)
+		require.NoError(t, err)
+		require.False(t, stepWitness.HasPreimage())
+
+		require.Equal(t, pc+4, state.PC) // PC must advance
+		require.Equal(t, uint64(0), state.Heap)
+		require.Equal(t, uint64(0), state.LoadReservation)
+		require.Equal(t, uint8(0), state.ExitCode) // ExitCode must be set
+		require.Equal(t, false, state.Exited)      // Must be exited
```
Comment (pcw109550) [inline rvgo/test/syscall_test.go:158]:
```suggestion
		require.Equal(t, false, state.Exited)      // Must not be exited
```
also other syscalls' comment must be fixed.

File: rvgo/test/syscall_test.go [hunk 2]
```diff
@@ -0,0 +1,903 @@
...
+			ExitCode:        0,
+			Exited:          false,
+			Memory:          fast.NewMemory(),
+			LoadReservation: 0,
+			Registers:       [32]uint64{17: fast.SysGetrlimit, 10: res, 11: addr},
+			Step:            0,
+		}
+		state.Memory.SetUnaligned(pc, syscallInsn)
+
+		fastState := fast.NewInstrumentedState(state, nil, os.Stdout, os.Stderr)
```
Comment (pcw109550) [inline rvgo/test/syscall_test.go:481]: We cannot test slow/EVM mode here yet, right?

File: rvgo/test/syscall_test.go [hunk 3]
```diff
@@ -0,0 +1,903 @@
...
+		_, err := fastState.Step(true)
+		require.Contains(t, err.Error(), "f0012")
```
Comment (pcw109550) [inline rvgo/test/syscall_test.go:483]: We should remove this hardcoded errorcode in the future.

**Review by pcw109550 — CHANGES_REQUESTED** [review]
Can you please rebase?

File: rvgo/test/syscall_test.go (hunk 2 above)
Comment (ImTei) [reply rvgo/test/syscall_test.go:482]: Yes. added TODO comment.

File: rvgo/test/syscall_test.go (hunk 3 above)
Comment (ImTei) [reply rvgo/test/syscall_test.go:484]: yeah we should. let me take it to the next task.

**Review by pcw109550 — APPROVED** [review]

**Pull request merged**